jax/_src/api.py

```diff
@@ -930,30 +930,32 @@ def _check_scalar(x):
 def _check_input_dtype_revderiv(name, holomorphic, allow_int, x):
   _check_arg(x)
   aval = core.get_aval(x)
-  if holomorphic:
-    if not dtypes.issubdtype(aval.dtype, np.complexfloating):
-      raise TypeError(f"{name} with holomorphic=True requires inputs with complex dtype, "
-                      f"but got {aval.dtype.name}.")
-  elif not allow_int and not (dtypes.issubdtype(aval.dtype, np.floating) or
-                              dtypes.issubdtype(aval.dtype, np.complexfloating)):
-    raise TypeError(f"{name} requires real- or complex-valued inputs (input dtype that "
-                    "is a sub-dtype of np.floating or np.complexfloating), "
-                    f"but got {aval.dtype.name}. If you want to use integer-valued "
-                    "inputs, use vjp or set allow_int to True.")
+  if (dtypes.issubdtype(aval.dtype, np.integer) or
+      dtypes.issubdtype(aval.dtype, np.bool_)):
+    if not allow_int:
```

Comment:
I cleaned this up, but wasn't sure whether `np.bool_` should be allowed or not.  It is allowed in the previous version.